core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java

```diff
@@ -300,6 +301,11 @@ void doReplace(Iterable<DataFile> deletedDataFiles, Iterable<DataFile> addedData
   private boolean isPartialFileScan(CombinedScanTask task) {
     if (task.files().size() == 1) {
       FileScanTask fileScanTask = task.files().iterator().next();
+      if (fileScanTask.file().format() == FileFormat.PARQUET) {
+        // in parquet format, there is an initial offset of 4 bytes
+        return fileScanTask.file().fileSizeInBytes() !=
+                (fileScanTask.length() + fileScanTask.file().splitOffsets().get(0));
```

Comment:
That sounds good. 
I will add `public static final int PARQUET_MAGIC_BYTES = 4;` in `Parquet` class, 
and use `Parquet.PARQUET_MAGIC_BYTES` instead of `fileScanTask.file().splitOffsets().get(0)`.
Is it ok?